lib/prompts/input.js

```diff
@@ -42,11 +43,14 @@ class InputPrompt extends Base {
     var bottomContent = '';
     var message = this.getQuestion();
     var transformer = this.opt.transformer;
+    var caretOffset;
 
     if (this.status === 'answered') {
       message += chalk.cyan(this.answer);
     } else if (transformer) {
       message += transformer(this.rl.line, this.answers);
+      caretOffset = message.slice(0, message.indexOf(constants.CARET));
```

Comment:
When not found, it'll do `slide(0, -1)`. That's probably not what we expected. Maybe `message.slice(0, Math.min(0, message.indexOf(constants.CARET)))` will fix this.

I wonder if that's causing the bug you talked about with regards to the caret position being offset by one.

Reply:
yes, the minor issue I found was what you mentioned. I didn't want to add IF statement for solving as it's a bug that could be ignored and I prefer code styling over solving it. But your solution using `Math.min()` is clever enough to maintain the code legibility.

Reply:
reviewing this it won't work with `Math.min()` just because min(0, -1) = -1. I'm working on a solution for it.

Reply:
`Math.max()` would work then; just provide the wrong function as example :P

Reply:
@SBoudrias it won't work with `Math.max()` either because `"foo-bar".slice(0, 0)` will strip the whole string. The only way it would work if with a function like this:
```javascript
message.slice(0, 
  ImaginaryPackage.positive(
    message.indexOf(constants.CARET), message.length)
)
```

That is, a function that will return the positive number of 2 numbers :)